1-Draft/RFCXXXX-RFC-Invoke-DscResource.md

```diff
@@ -0,0 +1,246 @@
+---
+RFC: RFCXXXX
+Author: Gael Colas
+Status: Draft
+SupersededBy: N/A
+Version: 0.9
+Area: Microsoft.PowerShell.DesiredStateConfiguration
+---
+
+# Invoke-DscResource
+
+Add cross-platform support for `Invoke-DscResource` in PowerShell 7+ without dependency on LCM and WMI.
+
+This RFC addresses the need to leverage the DSC ecosystem of resources from newer versions of PowerShell, the way to decouple the execution of resources from the LCM and CIM/WMI, and the user experience from a consumer and solution vendor/integrator point of view.
+
+## Motivation
+
+1. DSC has been cited by users and solution partners as a blocker for moving from Windows PowerShell to PowerShell 7+.
+
+2. As part of the latest survey on DSC usage started in June 2019, the top requested feature was support for Invoke-DscResource in PSCore without WMI dependency.
+
+3. The DSC Ecosystem and Community would greatly benefit from enabling DSC resources to be used in imperative scripts, in current user context.
```

Comment:
I'm not so sure that this is true (the greatly benefit part) - any source data for this or scenarios that would be enabled that are currently blocked?

Reply:
It opens the door to use DSC for more than local system configurations. I would have used DSC to configure load balancing, DNS, and vmware vm settings if I had the option. Effectively use it as a remote configuration manager.

Reply:
I'd add that this was the feedback we got from talking to vendors and users during PSConfEU, the DSC Survey and other ad hoc discussions.
Let's also bear in mind that no scenario is currently _covered_ for anything else than Windows PS 5.1 (which is only side by side with Core versions).

Also, this 3rd point specifically calls out benefits for the **Ecosystem **and **Community**:
- Lowering the bar to consume DSC Resources (without the need of specific permissions on a running node) should help draw more consumption from those who haven't looked at configuration Management. More usage = more opportunities to convert in the contributor funnel. More usage = stronger community = more leverage across the industry (i.e. pushing other products to create resources).
- It is encouraging better coding practices: idempotent scripting & a bit more declarative, in a more re-usable pattern than [what was buzzing not long ago](https://itnext.io/declarative-devops-30788ddd43cd).

Hit me up to discuss further ;)